### PR TITLE
Move repos search to redux

### LINKED ISSUE #330: [dashboard: add new repo]  Search through repos
---

Copied from Trello card: https://trello.com/c/YEILhCOS/123-dashboard-add-new-repo-search-through-repos
Labels: 

---

[Design specification](https://docs.google.com/document/d/1koTwI3cRr2S2vj5zHKKbR7aWxLRfShDVq2gHQ29cDyU/edit#bookmark=id.dd21jq6zcpfz): “Whenever there are any usable repos, opposite ‘Choose repos to add’ should be a search field…”

### PR DESCRIPTION
**Pull request by bartaz**
## Done

Moving repos search functionality out of component to redux store.

## QA

- Check out this feature branch
- Run the site using the command ```npm start -- --env=environments/dev.env```
- View the site locally in your web browser at: [http://0.0.0.0:8000/](http://0.0.0.0:8000/)
- Go to add repos, you should (still) be able to search through repos
- Run `npm test`, all tests should pass

## Issue / Card

Fixes #330

### CODE REVIEW

File: test/unit/src/common/components/select-repository-list/t_select-repository-list.js [hunk 1]
```diff
@@ -184,4 +185,22 @@ describe('<SelectRepositoryListComponent /> instance', function() {
       expect(spy).toHaveBeenCalledWith(addRepos(testProps.reposToAdd));
     });
   });
+
+  context('searching repositories', function() {
+    let wrapper;
+    let spy;
+
+    beforeEach(function() {
+      spy = spyOn(props, 'dispatch');
```
Comment (bartaz) [inline test/unit/src/common/components/select-repository-list/t_select-repository-list.js]: We don't need a spy here, as we don't dispatch any actions in this test.

Same applies to unnecessary `afterEach` block that clears the spy.

File: test/unit/src/common/actions/t_repositories.js [hunk 2]
```diff
@@ -108,23 +110,33 @@ describe('repositories actions', () => {
...
 
     it('should create a valid flux standard action', () => {
-      // expect(isFSA(action)).toBe(true);
+      const action = {
+        type: ActionTypes.REPOSITORIES_SEARCH,
+        payload: 'test'
+      };
+
+      store.dispatch(searchRepos('test'));
+      expect(isFSA(action)).toBe(true);
```
Comment (bartaz) [inline test/unit/src/common/actions/t_repositories.js]: We don't need to `dispatch` anything here (we are not even checking anything from the store).

We want to test if what `searchRepos` return is in Flux Standard Action format.

So: `expect(isFSA(searchRepos('test'))).toBe(true);` should be enough for whole test.

**Comment by richmccartney** [comment]
@bartaz everything should now be awesome, if you can give it another once over please 😄

**Comment by bartaz** [comment]
@richmccartney LGTM, thanks

**Pull request merged**